server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -417,81 +467,37 @@ private void loadAllLookupsAndInitStateRef()
     return lookups;
   }
 
-  private List<LookupBean> getLookupsListFromLookupConfig()
-  {
-    List<LookupBean> lookupBeanList;
-    if (lookupConfig.getEnableLookupSyncOnStartup()) {
-      String tier = lookupListeningAnnouncerConfig.getLookupTier();
-      lookupBeanList = getLookupListFromCoordinator(tier);
-      if (lookupBeanList == null) {
-        LOG.info("Coordinator is unavailable. Loading saved snapshot instead");
-        lookupBeanList = getLookupListFromSnapshot();
-      }
-    } else {
-      lookupBeanList = getLookupListFromSnapshot();
-    }
-    return lookupBeanList;
-  }
-
   private void startLookups(List<LookupBean> lookupBeanList)
   {
     ImmutableMap.Builder<String, LookupExtractorFactoryContainer> builder = ImmutableMap.builder();
     ExecutorService executorService = Execs.multiThreaded(
         lookupConfig.getNumLookupLoadingThreads(),
         "LookupReferencesManager-Startup-%s"
     );
-    ExecutorCompletionService completionService = new ExecutorCompletionService(executorService);
-    List<Future<Map.Entry>> futures = new ArrayList<>();
+    CompletionService<Map.Entry<LookupBean, LookupExtractorFactoryContainer>> completionService =
```

Comment:
if the final map deals solely with strings anyways, can the maps used elsewhere here also just use the `name` of the lookup bean? does it need to have the whole object?

Keeping the hash code and equals up to date is risky if using just the `name` is just as good

Reply:
Fixed